### PR TITLE
test: fix fmt pass, shorten warnings, clear SA1016

### LINKED ISSUE #7685: staticcheck: too many warnings in travis-ci
via https://travis-ci.org/coreos/etcd/jobs/219525242
```
Checking staticcheck...
egrep: auth/store_test.go:640:4: the goroutine calls T.Fatal, which must be called in the same goroutine as the test (SA2002)
auth/store_test.go:640:4: the goroutine calls T.Fatal, which must be called in the same goroutine as the test (SA2002)
contrib/raftexample/raftexample_test.go:105:3: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
e2e/ctl_v3_watch_test.go:72:3: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
e2e/etcd_release_upgrade_test.go:156:3: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
etcdctl/ctlv3/command/elect_command.go:76:36: os.Kill cannot be trapped (did you mean syscall.SIGTERM?) (SA1016)
...
```

### PR DESCRIPTION
**Pull request by heyitsanthony**
Fixes #7685

### CODE REVIEW

**Comment by heyitsanthony** [comment]
staticcheck update is causing failures in netutil, this also fixes that. PTAL

File: etcdctl/ctlv3/command/elect_command.go [hunk 1]
```diff
@@ -73,7 +74,7 @@ func observe(c *clientv3.Client, election string) error {
 
 	donec := make(chan struct{})
 	sigc := make(chan os.Signal, 1)
-	signal.Notify(sigc, os.Interrupt, os.Kill)
+	signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
```
Comment (fanminshi) [inline etcdctl/ctlv3/command/elect_command.go]: should we be consistent on what which pkg to use for signal?  use `syscall` for all signal?

File: etcdctl/ctlv3/command/elect_command.go [hunk 2]
```diff
@@ -107,7 +108,7 @@ func campaign(c *clientv3.Client, election string, prop string) error {
...
```
Comment (fanminshi) [inline etcdctl/ctlv3/command/elect_command.go]: use `syscall.SIGINT ` instead of `os.Interrupt`?

File: etcdctl/ctlv3/command/lock_command.go [hunk 3]
```diff
@@ -57,7 +58,7 @@ func lockUntilSignal(c *clientv3.Client, lockname string) error {
 	// unlock in case of ordinary shutdown
 	donec := make(chan struct{})
 	sigc := make(chan os.Signal, 1)
-	signal.Notify(sigc, os.Interrupt, os.Kill)
+	signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
```
Comment (fanminshi) [inline etcdctl/ctlv3/command/lock_command.go]: `syscall.SIGINT` instead of `os.Interrupt`?

**Comment by fanminshi** [comment]
lgtm

**Comment by heyitsanthony** [comment]
ppc64 failing on grpc leaks; merging

**Pull request merged**